statics/js/components/graph-layout.js

```diff
@@ -35,21 +35,24 @@ var LinkLabelBandwidth = Vue.extend({
     },
 
     updateData: function(link) {
-      if (!link.target.metadata.LastUpdateMetric) {
+      if (!link.target.metadata.LastUpdateMetric && !link.source.metadata.LastUpdateMetric) {
         return;
       }
 
       const defaultBandwidthBaseline = 1024 * 1024 * 1024; // 1 gbps
 
       link.bandwidthBaseline = (this.topology.bandwidth.bandwidthThreshold === 'relative') ?
-        link.target.metadata.Speed || defaultBandwidthBaseline : 1;
-
-      link.bandwidthAbsolute = this.bandwidthFromMetrics(link.target.metadata.LastUpdateMetric);
+        link.target.metadata.LastUpdateMetric ? link.target.metadata.Speed || defaultBandwidthBaseline : link.source.metadata.Speed || defaultBandwidthBaseline : 1;
```

Comment:
@hunchback ok with this ?

Reply:
Add comment - gist of it - extract metadata (from target or source) then simplify code - other than that logic seams ok

Reply:
yep, good idea to use an extracted metadata from source or target. will modify the code.